src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -1846,6 +1846,58 @@ void MergeTreeData::clearPartsFromFilesystemImpl(const DataPartsVector & parts_t
     }
 }
 
+size_t MergeTreeData::clearOldBrokenPartsFromDetachedDirecory()
+{
+    /**
+     * Remove old (configured by setting) broken detached parts.
+     * Only parts with certain prefixes are removed. These prefixes
+     * are such that it is guaranteed that they will never be needed
+     * and need to be cleared.
```

Comment:
It does not explain the difference between ctime and mtime. And does not mention the fact that ctime is updated on rename (at least on linux with ext4). Btw, is it true, for example, on FreeBSD/MacOS with zfs/btrfs/etc? Does POSIX provide some guarantees?

Reply:
(it's especially confusing, because `ctime` may look like "creation time")

Reply:
> Does POSIX provide some guarantees?

No:

> Some implementations mark for update the st_ctime field of renamed files and some do not. Applications which make use of the st_ctime field may behave differently with respect to renamed files unless they are designed to allow for either behavior.

https://pubs.opengroup.org/onlinepubs/009695399/functions/rename.html

Also explained here https://unix.stackexchange.com/a/211134

Reply:
Seems like we cannot rely on `st_ctime` :(